### ISSUE TITLE
Chromedriver console open with Pyinstaller

### ISSUE THREAD

**Issue by mrsachou** [ultrafunkamsterdam/undetected-chromedriver#623]
Hello, when i  compile with PyInstaller the Chromedriver console open also, i know with selenium4 now we can  hide the console like this :

```
from selenium import webdriver
from selenium.webdriver.chrome.service import Service as ChromeService # Similar thing for firefox also!
from subprocess import CREATE_NO_WINDOW # This flag will only be available in windows

# Define your own service object with the `CREATE_NO_WINDOW ` flag
# If chromedriver.exe is not in PATH, then use:
# ChromeService('/path/to/chromedriver')
chrome_service = ChromeService('chromedriver')
chrome_service.creationflags = CREATE_NO_WINDOW

driver = webdriver.Chrome(service=chrome_service)
```
But its not working with undetected chromedriver
Thanks for your help

**Comment by sebdelsol** [comment]
You're right you can't pass an already created service to `uc.Chrome()`. What makes it a little harder is that you'll need to pass to the service the actual executable_path that points to the patched chromedriver.

So you'll have to extensively modify `uc.Chrome()` class to make it work, you'll find it [**here** in \_\_init\_\_.py](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/master/undetected_chromedriver/__init__.py). 
In this file :
* You'll need first to import Selenium service  : `from selenium.webdriver.chrome.service import Service` 
* You'll need a new argument to the `uc.Chrome()` class : e.g `service_creationflags=None`
* You'll need to create the service and add the creation flag before initializing `uc.Chrome()` super class [**here**](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L401):
```python3
        if service_creationflags:
            service = Service(patcher.executable_path, port, service_args, service_log_path)
            service.creationflags = service_creationflags
        else:
            service = None
```

Usage:
```python3
from subprocess import CREATE_NO_WINDOW
import undetected_chromedriver as uc

if __name__ == "__main__":
    driver = uc.Chrome(service_creationflags=CREATE_NO_WINDOW)
```


Feel free to create a pull request for this.

**Comment by ThomasTrovon** [comment]
> You're right you can't forward an already created service to . What makes it a little harder is that you've to create the service with the actual that points to the patched chromedriver and that's done in .`uc.Chrome()``executable_path``uc.Chrome.__init__()`
> 
> So you'll have to actually modify class to make it works. It's defined in [this file **here**](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/master/undetected_chromedriver/__init__.py) :`uc.Chrome()`
> 
> * First you'll need to import Selenium service  :  [**here**](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L40).`from selenium.webdriver.chrome.service import Service`
> * And add a new kwarg to the  class : e.g  [**here**](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L104).`uc.Chrome()``service_creationflags=None`
> * Then create the service and set the  before initializing  parent class [**here**](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L400) :`creationflags``uc.Chrome()`
> 
> ```python
>         if service_creationflags:
>             service = Service(patcher.executable_path, port, service_args, service_log_path)
>             service.creationflags = service_creationflags
>         else:
>             service = None
> ```
> 
> * Finally forward the service argument to the  parent class [**here**](https://github.com/ultrafunkamsterdam/undetected-chromedriver/blob/b2e804e977c9cf3dd9c845474752cac142a72508/undetected_chromedriver/__init__.py#L409) :`uc.Chrome()`
> 
> ```python
>         super(Chrome, self).__init__(
>             executable_path=patcher.executable_path,
>             port=port,
>             options=options,
>             service_args=service_args,
>             desired_capabilities=desired_capabilities,
>             service_log_path=service_log_path,
>             keep_alive=keep_alive,
>             service=service,  # needed or the service will be re-created 
>         )
> ```
> 
> Usage:
> 
> ```python
> from subprocess import CREATE_NO_WINDOW
> import undetected_chromedriver as uc
> 
> if __name__ == "__main__":
>     driver = uc.Chrome(service_creationflags=CREATE_NO_WINDOW)
> ```
> 
> Here's the [>>**gist for it**<<](https://gist.github.com/sebdelsol/33efa1ea4593bdc1610e1139bb8af1aa). Please be aware it's based _on my own undetected-chromedriver fork_ in which I've added some evasions for headless mode among other things..
> 
> Feel free to create a pull request for this. I've tested it and it works, please tell me if you need more.

Bro... YOU ARE THE MAN! Worked perfectly! 

Pls @ultrafunkamsterdam add this in main version. s2 <3

**Comment by no-name-user-name** [comment]
driver = uc.Chrome(use_subprocess=True)

**Comment by sebdelsol** [comment]
I have added this feature in my [**fork**](https://github.com/sebdelsol/undetected-chromedriver) too.
`pip install git+https://github.com/sebdelsol/undetected-chromedriver@master`

**Comment by sebdelsol** [comment]
Note that Selenium has turned its `Service.creationflags` into `Service.creation_flags` since version 4.6.0.
So I have changed the relevant code **[here](https://github.com/sebdelsol/undetected-chromedriver/blob/2ef9791abb2b0c9d8bec8832897dd5c93a9dc4e3/undetected_chromedriver/__init__.py#L412)** in my fork. It works for any version of selenium.

**Comment by d0p0jake** [comment]
Hi, when I compile to .exe its still showing the blank console together with chrome. I use auto-py-to-exe as my .exe converter. Here is my code

`import undetected_chromedriver as uc
from subprocess import CREATE_NO_WINDOW
driver = uc.Chrome(service_creationflags=CREATE_NO_WINDOW, use_subprocess=True)
driver.get("https://www.samplesite.com/login")`

If I remove "use_subprocess=True" chrome won't launch at all.

**Comment by sebdelsol** [comment]
`service_creationflags` is not implemented on this version, so you can either wait for the relevant pull request to be merged or use my fork in the meantime.

`use_subprocess` is already set to `True` by default. If you set it to `False` then you'll have to protect your entry point and add `freeze_support()` [when you produce a Windows executable](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.freeze_support).
